tensorzero-core/src/config/mod.rs

```diff
@@ -758,6 +758,28 @@ impl Config {
         .await
     }
 
+    /// Load config from a ConfigSnapshot (historical config stored in ClickHouse)
+    pub async fn load_from_snapshot(
+        snapshot: ConfigSnapshot,
+        validate_credentials: bool,
+    ) -> Result<ConfigLoadInfo, Error> {
+        let config_load_info = if cfg!(feature = "e2e_tests") || !validate_credentials {
+            SKIP_CREDENTIAL_VALIDATION
+                .scope((), Self::load_from_toml(ConfigInput::Snapshot(snapshot)))
+                .await?
+        } else {
+            Self::load_from_toml(ConfigInput::Snapshot(snapshot)).await?
+        };
+
+        if validate_credentials {
```

Comment:
To avoid duplicating logic, could we move this into `load_from_toml` (which can take in the `validate_credentials` parameter)?